source/fuzz/transformation_wrap_vector_synonym.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021 Shiyu Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_WRAP_VECTOR_SYNONYM_H_
+#define SOURCE_FUZZ_TRANSFORMATION_WRAP_VECTOR_SYNONYM_H_
+
+#include <utility>
+
+#include "source/fuzz/data_descriptor.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/fuzz/protobufs/spirvfuzz_protobufs.h"
+#include "source/fuzz/transformation.h"
+#include "source/fuzz/transformation_composite_construct.h"
+#include "source/fuzz/transformation_context.h"
+#include "source/opt/instruction.h"
+#include "source/opt/ir_context.h"
```

Comment:
You should be able to do without several of these.

You should only include here the things that the .h file actually uses. So for example ir_context, transformation_context and spirvfuzz_protobufs are good. But transformation_composite_construct is definitely not good (it's not used in this header file) - I think some of the others are not good either.

Please think through each one and make a judgement on whether it's actually used in this header file.

It might be that some of these includes are needed only in the .cpp file.

This can be hard to check for manually. It's kind of a shame that spirv-tools doesn't use "include what you use" (iwyu), which helps figure this out automatically.

Reply:
Yes, only ``transformation.h`` and the ones you mentioned should be included in this file. And I think ``transformation_composite_construct.h`` should instead be included in the fuzzer pass. It seems like fuzzer pass is actually using the one included in here which clearly is not ideal.